ci/copy-canton.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+if [ "$#" -ne 1 ]; then
+  echo "Usage: $0 <canton_directory>"
+  exit 1
+else
+  canton_dir=$1
+fi
+
+#tmp=$(mktemp -d)
+#trap "rm -rf ${tmp}" EXIT
+#git -C $canton_dir worktree add -d $tmp
+#trap "git -C $canton_dir worktree remove -f $tmp" EXIT
+
+
+for path in community daml-common-staging README.md; do
+  rm -rf canton/$path
+  for f in  $(git -C "$canton_dir" ls-files $path); do
+    # we're only interested in copying files, not directories, as git-ls has
+    # explicitly expanded all directories
+    if [[ -f $canton_dir/$f ]]; then
+      # we create the parent directories of f under canton/ if they don't exist
+      mkdir -p canton/$(dirname $f)
```

Comment:
Oops, missed it. Thanks!

Reply:
Good point. What I need to know is where the root of the git repository is, so that I can modify the `canton` directory. I'm not sure I see how to achieve that with `DIR` above. Is there an idiomatic way of retrieving the root of the repo?

Reply:
`$DIR` with that invocation would be the directory in which this script is, so the folder you want to modify, assuming you don't move this file, is `$DIR/../canton` (and `$DIR` is `/path/to/git/project/ci`).

Reply:
I see, thanks! Done, and tested from two different directories.